src/Autocomplete/src/AutocompleteResultsExecutor.php

```diff
@@ -124,4 +118,19 @@ public function fetchResults(EntityAutocompleterInterface $autocompleter, string
 
         return new AutocompleteResults($results, $hasNextPage, $optgroups);
     }
+
+    /**
+     * @return array<string, mixed>
+     */
+    private static function getAutocompleterResult(EntityAutocompleterInterface $autocompleter, object $entity): array
```

Comment:
```suggestion
    /**
     * @return array{value: scalar, text: string}
     */
    private function getAutocompleterResult(EntityAutocompleterInterface $autocompleter, object $entity): array
```

I'm having doubt about the whole naming here. but in all cases let's use non-static methods here.

Reply:
Well yeah We  should rename this method formatResult() or something like buildResult().